Prepare environment for calico integration for Debian

ycointe opened the pull request:
**Component**: Buildchain

<!-- E.g. 'salt', 'containers', 'kubernetes', 'build', 'tests'... -->

**Context**: Metalk8s on Ubuntu 

**Summary**: Prepare the environment before to modify buildchain scrits to integrate calico for Debian.

**Acceptance criteria**: 
- Deploy cluster as usual.


---

<!-- Declare one or more issues to close once this PR gets merged -->

Closes: #1538 

<!-- If you want to refer to an issue while not closing it, use:

See: #ISSUE_NUMBER

-->

bert-e commented:
# Hello ycointe,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @teddyandrieux

* @monpote

* @jbertran

* @sayf-eddine-scality

* @nicolast

* @cmonfort

* @slaperche-scality

* @thomasdanan

* @zempashi

* @lucieleonard

* @gdemonet

* @carlito-scality

* @nootal

* @alexandre-allard-scality

ycointe committed:
Add some constant dedicated to Debian

Add some new constants dedicated to the Debian
environment.

ycointe committed:
Create Debian package root

Use the new constants, new package and repository target
to build and download packages in a Debian dedicated
folder.

slaperche-scality commented on buildchain/buildchain/packaging.py:
```diff
@@ -73,6 +73,12 @@ def task__package_mkdir_rpm_root() -> types.TaskDict:
+        directory=constants.PKG_DEB_ROOT, task_dep=['_package_mkdir_root']
+    ).task
+
 def task__package_mkdir_iso_root() -> types.TaskDict:
```
Something that we missed in the previous PR: this function should be renamed `task__package_mkdir_rpm_iso_root`

ycointe replied:
I don't understand, the task `task__package_mkdir_rpm_iso_root` exist 6 lines after. 
The tasks tagged with an OS familly like `rpm` or `deb` call the tasks untagged like, in this case, `task__package_mkdir_iso_root`
Please tell me if I missed something

slaperche-scality replied:
Meh, go cheated by the Github display:

![meh](https://user-images.githubusercontent.com/34654278/63430563-28da4680-c41d-11e9-8887-db05d3a83934.png)

Didn't see that there was a line gap…
Everything is fine, forget it ^^

slaperche-scality commented on packages/redhat/Dockerfile:
```diff
@@ -16,6 +16,9 @@ RUN yum install -y \
         rpmlint \
         yum-utils \
         && \
+    yum install -y \
```
Why running a new `yum install -y` when you could simply add `alien` to the list of installed package by the existing `yum install -y`?

ycointe replied:
Because during my tests, when I add `alien` with all other packages, it doesn't work.
I think it's a problem of dependencies uninstalled until all the previous packages are installed.
Add a new line with a new `yum install -y` is the only way i found to install `alien`

NicolasT replied:
`alien` sits in EPEL, not in the core repositories. Since we only install `epel-release` (which configures the EPEL repository on the system) in the first `yum install` invocation. Before that, EPEL is not available, so trying to install something from it fails.

So, not an issue and the second `yum install` invocation is fine (heck, there's no other way).

slaperche-scality replied:
Weird, what kind of error do you get?

slaperche-scality replied:
Ho ok, makes sense.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @teddyandrieux

* @monpote

* @jbertran

* @sayf-eddine-scality

* @nicolast

* @cmonfort

* @slaperche-scality

* @thomasdanan

* @zempashi

* @lucieleonard

* @gdemonet

* @carlito-scality

* @nootal

* @alexandre-allard-scality





The following reviewers are expecting changes from the author, or must review again:

* @slaperche-scality

NicolasT commented on buildchain/buildchain/constants.py:
```diff
@@ -33,12 +33,16 @@
 # Root of the RedHat repositories on the ISO.
 REPO_RPM_ROOT : Path = REPO_ROOT/'redhat'
+# Root of the Debian repositories on the ISO.
+REPO_DEB_ROOT : Path = REPO_ROOT/'debian'
```
So, is it Debian or is it Ubuntu? Those are not the same...

ycointe replied:
Debian, we sort by os-family.

ycointe committed:
Add Dockerfile to handle Debian packages

This Dockerfile permit to create a container
to build Debian packages and to convert some RPM
files in Debian.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @teddyandrieux

* @monpote

* @jbertran

* @sayf-eddine-scality

* @nicolast

* @cmonfort

* @slaperche-scality

* @thomasdanan

* @zempashi

* @lucieleonard

* @gdemonet

* @carlito-scality

* @nootal

* @alexandre-allard-scality

ycointe commented:
@bert-e approve

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @teddyandrieux

* @monpote

* @jbertran

* @sayf-eddine-scality

* @nicolast

* @cmonfort

* @slaperche-scality

* @thomasdanan

* @zempashi

* @lucieleonard

* @gdemonet

* @carlito-scality

* @nootal

* @alexandre-allard-scality










*The following options are set:* **approve**

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/metalk8s/commit/71867092630b6262d657d972ad6e37cfa0df4ce3) did not succeed in branch improvement/1538-calico-env-preparation.




*The following options are set:* **approve**

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.4`



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.4`



The following branches have **NOT** changed:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`



Please check the status of the associated issue None.

Goodbye ycointe.
